lib/rbac/filterer.rb

```diff
@@ -417,17 +402,33 @@ def scope_to_tenant(scope, user, miq_group)
       tenant_id_clause ? scope.where(tenant_id_clause) : scope
     end
 
+    ##
+    # Main scoping method
+    #
     def scope_targets(klass, scope, rbac_filters, user, miq_group)
+      # Results are scoped by tenant if the TenancyMixin is included in the class,
+      # with a few manual exceptions (User, Tenant). Note that the classes in
+      # TENANT_ACCESS_STRATEGY are a consolidated list of them.
       if klass.respond_to?(:scope_by_tenant?) && klass.scope_by_tenant?
         scope = scope_to_tenant(scope, user, miq_group)
       end
 
-      if apply_rbac_to_class?(klass)
-        scope = scope_by_direct_rbac(scope, rbac_filters, user, miq_group)
-      elsif apply_rbac_to_associated_class?(klass)
-        scope = scope_by_indirect_rbac(scope, rbac_filters, user, miq_group)
-      elsif apply_user_group_rbac_to_class?(klass, miq_group)
-        scope = scope_by_user_group_rbac(scope, user, miq_group)
+      if apply_rbac_directly?(klass)
+        filtered_ids = calc_filtered_ids(scope, rbac_filters, user, miq_group)
```

Comment:
> We have the goal of having it alter the scope instead.

We *really* need to have a place to write this stuff down; I feel like there are many [good] goals you have that I'm completely unaware of and I don't want there to be confusion about where we'd both like to go with improving RBAC in the future. 💙